src/components/OAuth/OAuth.type.ts

```diff
@@ -0,0 +1,45 @@
+import { SetURLSearchParams } from "react-router-dom";
+
+import { AppDispatch } from "@/src/redux/store";
+import { IUserStore } from "@/src/redux/types/user";
+
+export enum OAuthApiEndPoints {
+  GOOGLE_TOKEN = "/token",
+}
+
+export enum OAuthVariant {
+  FACEBOOK = "facebook",
+  GOOGLE = "google",
+  TWITTER = "twitter",
+}
+
+export type OAuthProps = {
+  className?: string;
+  text?: string;
+  buttonColor?: "primary" | "secondary";
+  buttonVariant?: "outlined" | "contained" | "text";
```

Comment:
you can use my types from Button.type for buttonVariant etc

Reply:
done